browser/js/dashboard/gridbox/widgetSettingsCtrl.js

```diff
@@ -1,5 +1,5 @@
-  app.controller('WidgetSettingsCtrl', ['$scope', 'DashboardFactory', '$timeout', '$rootScope', '$uibModalInstance', 'widget', 'GeneratorFactory',
-    function($scope, DashboardFactory, $timeout, $rootScope, $uibModalInstance, widget, GeneratorFactory) {
+app.controller('WidgetSettingsCtrl', ['$scope', 'DashboardFactory', '$timeout', '$rootScope', '$modalInstance', 'widget', 'GeneratorFactory', '$interval', 'WidgetSettingsFactory', '$uibModalInstance',
+    function($scope, DashboardFactory, $timeout, $rootScope, $modalInstance, widget, GeneratorFactory, $interval,WidgetSettingsFactory, $uibModalInstance) {
       $scope.widget = widget;
       if(widget.chart.data && widget.chart.data[0].values.length){
         $scope.dataKeys = Object.keys(widget.chart.data[0].values[0]);
```

Comment:
Aren't you retrieving the keys in your factory, [here](https://github.com/mosane/flashdash/pull/61/files#diff-a75321a86cef9ad2f210611277891c6fR16)?